Modules/Video/Core/include/itkImageToVideoFilter.h

```diff
@@ -0,0 +1,159 @@
+  using OutputFrameSpatialRegionType = typename Superclass::OutputFrameSpatialRegionType;
+  using OutputTemporalRegionType = typename TOutputVideoStream::TemporalRegionType;
+
+  /** Input type alias */
+  using InputImagePointer = typename InputImageType::Pointer;
+  using InputImageConstPointer = typename InputImageType::ConstPointer;
+  using InputImageRegionType = typename InputImageType::RegionType;
+  using InputImagePixelType = typename InputImageType::PixelType;
+  using InputImageIndexType = typename InputImageType::IndexType;
+  static constexpr unsigned int InputImageDimension = TInputImage::ImageDimension;
+
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(ImageToVideoFilter, VideoSource);
+
+  /** Set the input Image for this process object */
+  using Superclass::SetInput;
+  virtual void
+  SetInput(const InputImageType * image);
+
+  virtual void
+  SetInput(unsigned int idx, const InputImageType * videoStream);
+
+  /** Get the input Image for this process object */
+  const InputImageType *
+  GetInput() const;
+
+  const InputImageType *
+  GetInput(unsigned int idx) const;
+
+  /** Allow the user to specify the axis in the input image that will
+   *  correspond to the temporal axis in the output temporal object. */
+  itkGetMacro(FrameAxis, typename itk::IndexValueType);
+  itkSetMacro(FrameAxis, typename itk::IndexValueType);
+
+protected:
+  ImageToVideoFilter();
+  ~ImageToVideoFilter() override = default;
+
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+  /** Get a non-const version of the input for internal use when setting
+   * input's requested regions. This is the only time input should be modified
+   */
+  InputImageType *
+  GetInput();
+
+  InputImageType *
+  GetInput(unsigned int idx);
+
+  /** Set up the output VideoStream via spatial and temporal regions
+   * derived from the spatial regions of the input Image.
+   */
+  void
+  GenerateOutputInformation() override;
+
+  /** Override the Superclass::UpdateOutputInformation method
+   * so that the temporal output region is defined entirely within
+   * the GenerateOutputInformation method.
+   */
+  void
+  UpdateOutputInformation() override;
+
+  /** Generate the requested regions in the output VideoStream
+   *  from the size of the available input Image.
+   */
+  void
+  GenerateOutputRequestedRegion(DataObject * output) override;
+
+  /** Override the default implementation of GenerateInputRequestedRegion from
+   * VideoSource so that we only get spatial regions from the image input.
+   */
+  void
+  GenerateInputRequestedRegion() override;
+
+  /** Graft pixel data from input image onto output video frames */
+  void
+  GenerateData() override;
+
+private:
+  /** Index representing the axis to use for slicing the input Image
+   *  into frames for the output VideoStream.
+   */
+  typename itk::IndexValueType m_FrameAxis{ 0 };
```

Comment:
remove `itk::`

Is a better default `TInputImage::ImageDimension - 1`?

Reply:
Is typename needed?

Reply:
@aylward `typename itk::` is not needed here, thanks for the catch!

@thewtex could you explain more on the default value? My intuition is that the slowest-moving axis (`0`) would be the most likely to represent the temporal axis as frames are added to a static data set. Is that incorrect, or are there situations where the fastest-moving axis (`TInputImage::ImageDimension - 1`) would be a better choice?

Reply:
I think this goes back to what we are considering the identify transformation matrix to mean and does it jive with what we are currently doing.  Like if I use the existing IO machinery to output a .nii file (with time) what happens?

Reply:
> @thewtex could you explain more on the default value? My intuition is that the slowest-moving axis (0) 

My assumption was that `TInputImage::ImageDimension - 1` is the slowest moving axis, but that is not correct?

Reply:
> My assumption was that TInputImage::ImageDimension - 1 is the slowest moving axis

Based on [`itk::ImageBase::ComputeOffsetTable`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkImageBase.hxx#L184) the consensus here is correct that the 0th index is the fastest-moving in `itk::Image`. I had trouble finding documentation on the pixel buffer memory layout, if it exists could someone please link it here?

I will revise the default axis here as well as the related test to match axis order.

Thanks for catching this @thewtex !